align/pdk/finfet/transistor_array.py

```diff
@@ -31,33 +31,34 @@ def mos_array_temporary_wrapper(self, x_cells, y_cells, pattern, vt_type, ports,
 
         #################################################################################################
         # TODO: All of below goes away when TransistorArray is passed to mos_array as shown below
-        for key in ['m', 'real_inst_type']:
+        for key in ['M', 'real_inst_type']:
             assert key in parameters, f'Missing transistor parameter {key}'
-        assert 'nf' or 'stack' in parameters, f'Missing transistor parameter nf or stack'
+        assert 'NF' or 'STACK' in parameters, f'Missing transistor parameter nf or stack'
 
-        if 'stack' in parameters:
-            nf = 'stack'
+        if 'STACK' in parameters and int(parameters['STACK']) > 1:
+            nf = 'STACK'
             device_type = 'stack'
-        elif 'nf' in parameters:
-            nf = 'nf'
+        elif 'NF' in parameters and int(parameters['NF']) > 1:
+            nf = 'NF'
             device_type = 'parallel'
         else:
             nf = device_type = None
-            assert False, f'Either nf or stack parameter should be defined {parameters}'
-
-        if 'w' in parameters:
-            nfin = parameters['w'] * 1e10 // self.pdk['Fin']['Pitch']
-            # w in the netlist is the effective total width for a single transistor 
-            nfin = nfin // parameters[nf]
-        elif 'nfin' in parameters:
-            nfin = parameters['nfin']
+            assert False, f'Either nf>1 or stack>1 parameter should be defined {parameters}'
+
+        if 'W' in parameters:
+            nfin = int(float(parameters['W']) * 1e10) // self.pdk['Fin']['Pitch']
```

Comment:
What should happen here if 'W' is not divisible by the fin pitch? Currently is silently truncates. It should probably be an error or maybe round to nearest. Truncate seems wrong.

Reply:
This is a hack added by Soner for intel 22fl pdk.

Reply:
yes, this was intentional as the parsed floating point numbers had long tail.

Reply:
You'll probably want to replace this with:
```
w = round(float(parameters['W']) * 1e10)
nfin,r = divmod(w, self.pdk['Fin']['Pitch'])
if r != 0:
   logger.warning( f"nfin not integral: {w} // {self.pdk['Fin']['Pitch']} = {nfin} with remainder {r}")
```
Otherwise 2.9999 might round to 2 instead of 3 and also w in anstroms might now be a multiple of the fin pitch [if we are worried that W is not an integral number of angstroms, we need to do more here as well.]
We could make this check be in a schema validator.